webapp/boards/src/types/mattermost-webapp/index.d.ts

```diff
@@ -14,7 +14,7 @@ export interface PluginRegistry {
     registerCustomRoute(route: string, component: ReactResolvable): any
     registerProductRoute(route: string, component: ReactResolvable): any
     unregisterComponent(componentId: string): any
-    registerProduct(baseURL: string, switcherIcon: string, switcherText: string, switcherLinkURL: string, mainComponent: ReactResolvable, headerCentreComponent: ReactResolvable, headerRightComponent: ReactResolvable, showTeamSidebar: boolean): any
+    registerProduct(baseURL: string, switcherIcon: string, switcherText: string, switcherLinkURL: string, mainComponent: ReactResolvable, publicComponent: ReactResolvable | null, headerCentreComponent: ReactResolvable, headerRightComponent: ReactResolvable, showTeamSidebar: boolean): any
```

Comment:
Can we add this to the end of the function signature instead, to make it less likely to break something? I imagine this affects Playbooks, and potentially community plugins. I know we advise not to use this `registerProduct` method, but I think we should try to make it backwards compatible when possible

Reply:
Right, addressed on https://github.com/mattermost/mattermost-server/pull/22710/commits/fab9018f0fd722c52d1075b4b0eff17be667b2a7 and updated the Playbooks call to explicitly pass an argument on https://github.com/mattermost/mattermost-server/pull/22710/commits/a41eb27793d8146d781de781d3588a407b755c94. Thanks!